### PR TITLE
Improve YOLOv9 performance

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

This PR includes:
- [x] Update hyperparameters and configurations for performance improvement
- [x] Fix loss weights
- [x] Update export option to align performance of torch and ov IR
- [x] Enable to skip current batch without ground truth
    - This feature is because yolov9 loss cannot calculate loss values without ground truth

**The below table shows F1-score on pothole_medium before and after this update and the score on bdd_medium.
| model | before (pothole_medium) | after (pothole_medium) | after (bdd_medium) |
| -- | -- | -- | -- |
| v9_s | 0.490 | 0.737 (+0.247) | 0.513 |
| v9_m | 0.442 | 0.706 (+0.264) | 0.558 |
| v9_c | 0.255 | 0.731 (+0.475) | 0.560 |

**The below table shows F1-score consistency between torch and ov IR models after this update.
| model | dataset | type | mAP | F1-score |
| -- | -- | -- | -- | -- |
| v9_s | pothole_medium | torch | 0.343 | 0.737 |
| | | ov | 0.402 | 0.754 |
| | | ov (int8) | 0.399 | 0.748 |
| | bdd_medium | torch | 0.142 | 0.513 |
| | | ov | 0.182 | 0.517 |
| | | ov (int8) | 0.167 | 0.506 |
| v9_m | pothole_medium | torch | 0.335 | 0.706 |
| | | ov | 0.368 | 0.725 |
| | | ov (int8) | 0.375 | 0.709 |
| | bdd_medium | torch | 0.170 | 0.558 |
| | | ov | 0.189 | 0.556 |
| | | ov (int8) | 0.190 | 0.550 |
| v9_c | pothole_medium | torch | 0.377 | 0.731 |
| | | ov | 0.413 | 0.747 |
| | | ov (int8) | 0.436 | 0.766 |
| | bdd_medium | torch | 0.202 | 0.560 |
| | | ov | 0.215 | 0.572 |
| | | ov (int8) | 0.222 | 0.560 |

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by harimkang — DISMISSED** [review]

**Review by eunwoosh — DISMISSED** [review]

**Review by kprokofi — DISMISSED** [review]

**Comment by sungchul1** [comment]
~This PR will be updated more soon, don't merge it.~

File: src/otx/algo/detection/losses/yolov9_loss.py [hunk 1]
```diff
@@ -405,6 +405,10 @@ def forward(
         Returns:
             dict[str, Tensor]: The loss dictionary.
         """
+        if targets.shape[1] == 0:
```
Comment (eunwoosh) [inline src/otx/algo/detection/losses/yolov9_loss.py:408]: Could you explain what it's for?

Comment (sungchul1) [reply src/otx/algo/detection/losses/yolov9_loss.py:408]: `targets` has (batch, num_bboxes, 5).

Comment (kprokofi) [reply src/otx/algo/detection/losses/yolov9_loss.py:408]: Won't it destroy the training? Can't we just raise an error?

Comment (sungchul1) [reply src/otx/algo/detection/losses/yolov9_loss.py:408]: I added below to skip the iteration
https://github.com/openvinotoolkit/training_extensions/blob/2fb694e3d9907e2dd3ecb9442ba2184166361758/src/otx/core/model/base.py#L146-L150

Comment (sungchul1) [reply src/otx/algo/detection/losses/yolov9_loss.py:408]: This part doesn't affect other tasks and other models because there is no model that returns None as loss output.

Comment (eunwoosh) [reply src/otx/algo/detection/losses/yolov9_loss.py:408]: Thanks for explanation. Could you add unit test for this case?

Comment (kprokofi) [reply src/otx/algo/detection/losses/yolov9_loss.py:408]: Is it expected that model returns None sometimes?

Comment (sungchul1) [reply src/otx/algo/detection/losses/yolov9_loss.py:408]: > Is it expected that model returns None sometimes?

Just for yolov9. If other models could return None, TypeError would be raised.
https://github.com/openvinotoolkit/training_extensions/blob/2fb694e3d9907e2dd3ecb9442ba2184166361758/src/otx/core/model/base.py#L181

Comment (sungchul1) [reply src/otx/algo/detection/losses/yolov9_loss.py:408]: > Thanks for explanation. Could you add unit test for this case?

Done at dce91cd

**Review by harimkang — DISMISSED** [review]

**Review by eunwoosh — DISMISSED** [review]

**Review by harimkang — DISMISSED** [review]

**Review by eunwoosh — DISMISSED** [review]

**Review by eunwoosh — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

**Comment by kprokofi** [comment]
detection tests are passed 
image

**Comment by sovrasov** [comment]
@kprokofi thanks for arranging the tests. Merging without the queue, since it doesn't work
